scripts/ui.ts

```diff
@@ -111,47 +113,69 @@ async function requestLoginAccess(permissions: Permission[]): Promise<[boolean,
     throw new Error(reason);
   }
 
+  // Get the seed and email.
+
   let seed = checkStoredSeed();
 
+  // If we don't have a seed, get it from the seed provider. We will also get
+  // the email if the user is signing up for the first time.
   if (!seed) {
-    // If we don't have a seed, show seed provider chooser.
+    // Show seed provider chooser.
 
     const seedProviderDisplayUrl = await getSeedProviderDisplayUrl();
 
     // User has chosen seed provider, open seed provider display.
 
     log("Calling runSeedProviderDisplay");
-    seed = await runSeedProviderDisplay(seedProviderDisplayUrl);
+    let email = null;
+    [seed, email] = await runSeedProviderDisplay(seedProviderDisplayUrl);
+
+    // Register and get the JWT.
+    let jwt = null;
+    if (email) {
+      try {
+        jwt = await register(client, seed, email);
+      } catch (e1) {
+        try {
+          jwt = await login(client, seed, email);
+        } catch (e2) {
+          throw new Error(`Could not register: ${e1}. Could not login: ${e2}`);
+        }
+      }
+    }
```

Comment:
Does this mean that if we don't have the email we won't try to login/register? Because we don't set the email on login.

I'm a bit hazy on how this will be used because while the method is registered in the `init` function, I don't see it being used anywhere. Maybe it's being called implicitly and I don't see that.

Reply:
`requestLoginAccess` is called by skapps and triggers the MySky UI.

Reply:
Could you elaborate on that in the context of the quest? What happens when the MySky UI is triggered - does it initialise the email somehow or not?

Reply:
Apologies for the vague answer. The method `requestLoginAccess` is called from skapps using a handshake to the MySky UI window. (`init` sets up this handshake, providing the available methods.) The MySky UI is the only visual part of MySky right now and it includes login/signup as well as permissions. Hope that answered your questions.